shared/middleware/helpers/auth_helpers.rb

```diff
@@ -0,0 +1,28 @@
+#
+# Utility methods that help middleware access dashboard authentication and
+# permissions information.
+#
+
+# @returns [Integer] the user_id associated with the current request
+def current_user_id
+  @dashboard_request ||= Rack::Request.new(@env)
```

Comment:
where is @env defined? Generally, it would be helpful to have a comment explaining the steps you're taking to make this work in both dashboard and pegasus.